src/displayapp/screens/Calculator.cpp

```diff
@@ -0,0 +1,374 @@
+    
+    char op;
+    
+    double calculate() override {
+      double rightVal = right->calculate();
+      double leftVal = left->calculate();
+      switch (op) {
+        case '^':
+          return pow(leftVal, rightVal);
+        case 'x':
+          return leftVal * rightVal;
+        case '/':
+          if (rightVal == 0.0) {
+            errno = EDOM;
+            return 0.0;
+          }
+          return leftVal / rightVal;
+        case '+':
+          return leftVal + rightVal;
+        case '-':
+          return leftVal - rightVal;
+      }
+      errno = EINVAL;
+      return 0.0;
+    };
+    
+  };
+  
+  uint8_t getPrecedence(char op) {
+    switch (op) {
+      case '^':
+        return 4;
+      case 'x':
+      case '/':
+        return 3;
+      case '+':
+      case '-':
+        return 2;
+    }
+    return 0;
+  }
+  
+  bool leftAssociative(char op) {
+    switch (op) {
+      case '^':
+        return false;
+      case 'x':
+      case '/':
+      case '+':
+      case '-':
+        return true;
+    }
+    return false;
+  }
+  
+}
+
+
+static void eventHandler(lv_obj_t* obj, lv_event_t event) {
+  auto calc = static_cast<Calculator*>(obj->user_data);
+  calc->OnButtonEvent(obj, event);
+}
+
+Calculator::~Calculator() {
+  lv_obj_clean(lv_scr_act());
+}
+
+static const char* buttonMap1[] = {"7", "8", "9", "/", "\n",
+                                   "4", "5", "6", "x", "\n",
+                                   "1", "2", "3", "-", "\n",
+                                   ".", "0", "=", "+", "",};
+
+static const char* buttonMap2[] = {"7", "8", "9", "(", "\n",
+                                   "4", "5", "6", ")", "\n",
+                                   "1", "2", "3", "^", "\n",
+                                   ".", "0", "=", "+", "",};
+
+Calculator::Calculator(DisplayApp* app, Controllers::MotorController& motorController)
+    : Screen(app), motorController{motorController} {
+  result = lv_label_create(lv_scr_act(), nullptr);
+  lv_label_set_long_mode(result, LV_LABEL_LONG_BREAK);
+  lv_label_set_text(result, "0");
+  lv_obj_set_size(result, 180, 60);
+  lv_obj_set_pos(result, 0, 0);
+  
+  returnButton = lv_btn_create(lv_scr_act(), nullptr);
+  lv_obj_set_size(returnButton, 52, 52);
+  lv_obj_set_pos(returnButton, 186, 0);
+  lv_obj_t* returnLabel;
+  returnLabel = lv_label_create(returnButton, nullptr);
+  lv_label_set_text(returnLabel, "<=");
+  lv_obj_align(returnLabel, nullptr, LV_ALIGN_CENTER, 0, 0);
+  returnButton->user_data = this;
+  lv_obj_set_event_cb(returnButton, eventHandler);
+  
+  buttonMatrix = lv_btnmatrix_create(lv_scr_act(), nullptr);
+  lv_btnmatrix_set_map(buttonMatrix, buttonMap1);
+  lv_obj_set_size(buttonMatrix, 240, 180);
+  lv_obj_set_pos(buttonMatrix, 0, 60);
+  lv_obj_set_style_local_pad_all(buttonMatrix, LV_BTNMATRIX_PART_BG, LV_STATE_DEFAULT, 0);
+  buttonMatrix->user_data = this;
+  lv_obj_set_event_cb(buttonMatrix, eventHandler);
+  running = true;
```

Comment:
The running variable doesn't need to be initialized or defined in the header, because it's already done in Screen.h